homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -120,6 +121,7 @@
             vol.Optional(CONF_HS_VALUE_TEMPLATE): cv.template,
             vol.Optional(CONF_MAX_MIREDS): cv.positive_int,
             vol.Optional(CONF_MIN_MIREDS): cv.positive_int,
+            vol.Optional(CONF_ICON): cv.icon,
```

Comment:
Shouldn't this be part of a higher schema to make this behavior consistent across all platforms provided by the MQTT integration?

Reply:
I agree, however, I think refactoring should be a separate PR including other common fields as well, not just the icon.
For now, this is consistent with existing MQTT components (and some others as well).

Reply:
I agree with @frenck, let's make this a general schema.
Please refactor the PR to add a new mqtt mixin, and add that to the mqtt components already support CONF_ICON + light.

Check homeassistant/components/mqtt/mixins.py

Reply:
I was just checking into `MqttEntity` as well and how `CONF_UNIQUE_ID` is implemented. Will do that!

Reply:
@emontnemery I have now hoisted the name property, although i kept the `CONF_NAME` property inside individual files as they contained default values.

Reply:
Forgot to remove the original name functions, gonna remove them soon

Reply:
Are you planning to do more changes to the PR?

Reply:
@emontnemery I see that CONF_UNIQUE_ID is set only in `MqttEntity` initializer, but not updated via `discovery_update()`. Is this intentional?

Also, I see a lot of `config.get()` usage without a default parameter..
Should refactor them as well to take advantage of `KeyError`?

Reply:
Ufa

Reply:
@jking95 ?

Reply:
> @emontnemery I see that CONF_UNIQUE_ID is set only in MqttEntity initializer, but not updated via discovery_update(). Is this intentional?

Yes, it's intentional, it seems like an unnecessary complication to support changing unique_id's as the suggested unique_id may be rejected.